nnstreamer/tensor_filter/tensor_filter_nntrainer.cc

```diff
@@ -45,110 +46,72 @@ static const gchar *nntrainer_accl_support[] = {NULL};
 void init_filter_nntrainer(void) __attribute__((constructor));
 void fini_filter_nntrainer(void) __attribute__((destructor));
 
-NNTrainerInference::NNTrainerInference(const char *model_config_,
-                                       const GstTensorFilterProperties *prop) {
-  gst_tensors_info_init(&inputTensorMeta);
-  gst_tensors_info_init(&outputTensorMeta);
+static std::unique_ptr<GstTensorInfo>
+to_nnst_tensor_dim(const nntrainer::TensorDim &dim) {
+  auto info = std::unique_ptr<GstTensorInfo>(g_new(GstTensorInfo, 1));
+  gst_tensor_info_init(info.get());
 
-  model_config = g_strdup(model_config_);
-  loadModel();
-  model->compile();
-  model->initialize();
-
-  validateTensor(&prop->input_meta, true);
-  validateTensor(&prop->output_meta, false);
-
-  model->readModel();
-
-  gst_tensors_info_copy(&inputTensorMeta, &prop->input_meta);
-  gst_tensors_info_copy(&outputTensorMeta, &prop->output_meta);
-}
-
-NNTrainerInference::~NNTrainerInference() {
-  if (model != nullptr) {
-    delete model;
+  info->type = _NNS_FLOAT32;
+  for (unsigned int i = 0; i < NUM_DIM; ++i) {
+    info->dimension[i] = dim.getTensorDim(NUM_DIM - i - 1);
   }
 
-  gst_tensors_info_free(&inputTensorMeta);
-  gst_tensors_info_free(&outputTensorMeta);
-  g_free(model_config);
+  return info;
 }
 
-const char *NNTrainerInference::getModelConfig() { return model_config; }
-
-void NNTrainerInference::validateTensor(const GstTensorsInfo *tensorInfo,
-                                        bool is_input) {
-
-  nntrainer::TensorDim dim;
-  nntrainer_tensor_info_s info_s;
-
-  if (is_input)
-    dim = model->getInputDimension()[0];
-  else
-    dim = model->getOutputDimension()[0];
-
-  if (tensorInfo->info[0].type != _NNS_FLOAT32)
-    throw std::invalid_argument(
-      "only float32 is supported for input and output");
-
-  info_s.rank = NUM_DIM;
-
-  info_s.dims.push_back(tensorInfo->info[0].dimension[NUM_DIM - 1]);
-
-  // Here, we only compare channel, height, width.
-  // Just use batch from info variable, cause it will be updated if it differs.
-  for (unsigned int i = 1; i < NUM_DIM; ++i) {
-    if (tensorInfo->info[0].dimension[i - 1] != dim.getDim()[NUM_DIM - i])
-      throw std::invalid_argument("Tensor dimension doesn't match");
+static nntrainer::TensorDim to_nntr_tensor_dim(const GstTensorInfo *info) {
+  auto d = info->dimension;
```

Comment:
it might be good to put the actual type here, and consider &?

Reply:
Done